pkg/proxy/iptables/proxier.go

```diff
@@ -538,30 +578,33 @@ func (proxier *Proxier) SyncLoop() {
 
 func (proxier *Proxier) OnServiceAdd(service *api.Service) {
 	namespacedName := types.NamespacedName{Namespace: service.Namespace, Name: service.Name}
-	proxier.serviceChanges.update(&namespacedName, nil, service)
-
-	proxier.syncProxyRules(syncReasonServices)
+	if proxier.serviceChanges.update(&namespacedName, nil, service) {
+		// TODO: Check if endpoints and services are synced.
```

Comment:
Sorry. What I meant here is that if initial syncing of the informer fo either endpoints or services is not finished (this is happening exactly once after kube-proxy restart), then we are not calling syncProxyRules, to avoid situation with updating it with partial state (if it was correct before). 

So what we can loose without solving this TODO is that we can unnecessarily call syncProxyRules when kube-proxy is just starting and loose a tocken. But that only after kube-proxy (re)start so it's not that critical I guess.

For now I clarified the TODO. I can solve it in a separate PR.